src/core/protocol-authorization.ts

```diff
@@ -276,20 +276,13 @@ export class ProtocolAuthorization {
       return;
     }
 
-    const protocolRole = (incomingMessage as RecordsRead).message.descriptor.protocolRole;
+    const protocolRole = (incomingMessage as RecordsRead).authorizationPayload?.protocolRole;
 
     // Only verify role if there is a role being invoked
     if (protocolRole === undefined) {
       return;
     }
 
-    if (incomingMessage.author === undefined) {
```

Comment:
For my understanding: it's not immediately obvious to me why author can be `undefined` under protocol authorization, please educate me! Is it worthwhile to add a test for protocol auth with author being `undefined`?

Reply:
It is valid for an authorless `RecordsRead` to read a record with rule `who: 'anyone', can: 'read'`. In this case, it's always invalid for an incoming `RecordsRead` invokes a protocol role, so we validated that `author !== undefined`. We can remove this check because we moved `protocolRole` to `authorization`. Now, if `protocolRole` is present then `author` must also be present.

Reply:
> if `protocolRole` is present then `author` must also be present.

Gold, thank you!